### ISSUE TITLE
`MeanSquaredError` throws `FutureWarning` while passing `squared=True`

### ISSUE THREAD

**Issue by yarnabrina** [sktime/sktime#5874]
## Code

```python
>>> 
>>> from sktime.performance_metrics.forecasting import MeanSquaredError
>>> rmse = MeanSquaredError(square_root=True)
>>> 
>>> import numpy
>>> y_true = numpy.array([[0.5, 1], [-1, 1], [7, -6]])
>>> y_pred = numpy.array([[0, 2], [-1, 2], [8, -5]])
>>> 
>>> rmse(y_true, y_pred)
/home/anirban/conda-environments/sktime/lib/python3.10/site-packages/sklearn/metrics/_regression.py:483: FutureWarning: 'squared' is deprecated in version 1.4 and will be removed in 1.6. To calculate the root mean squared error, use the function'root_mean_squared_error'.
  warnings.warn(
0.8227486121839513
>>> 
```

## Environment

Ubuntu WSL, python 3.10, sktime 0.26.0

## What to do

1. new dedicated metric for RMSE
2. if-else in the existing `mean_squared_error` function

**Comment by fkiraly** [comment]
2 is more deprecation safe, imo? `if not squared then np.sqt`

**Comment by yarnabrina** [comment]
I also prefer 2, but I actually meant in this sense. Both are probably identical.

```python
from sklearn.metrics import mean_squared_error as _mean_squared_error
from sklearn.metrics import root_mean_squared_error as _root_mean_squared_error

metric = _root_mean_squared_error if square_root else _mean_squared_error
return metric(y_true, y_pred, sample_weight=horizon_weight, multioutput=multioutput)
```

**Comment by fkiraly** [comment]
hm, I am slightly apprehensive since the architecture currently uses delegation with a fixed attribute being the metric, set in the class or constructor. So the `__init__` may have to do sth here.
